core/src/main/scala/org/apache/spark/BarrierCoordinator.scala

```diff
@@ -225,10 +262,19 @@ private[spark] sealed trait BarrierCoordinatorMessage extends Serializable
  * @param stageAttemptId ID of current stage attempt
  * @param taskAttemptId Unique ID of current task
  * @param barrierEpoch ID of the `barrier()` call, a task may consist multiple `barrier()` calls.
+ * @param requestMethod The BarrierTaskContext method that was called to trigger BarrierCoordinator
+ * @param allGatherMessage Message sent from the BarrierTaskContext if requestMethod is ALL_GATHER
  */
 private[spark] case class RequestToSync(
     numTasks: Int,
     stageId: Int,
     stageAttemptId: Int,
     taskAttemptId: Long,
-    barrierEpoch: Int) extends BarrierCoordinatorMessage
+    barrierEpoch: Int,
```

Comment:
Maybe we can have `BarrierRequest` and `AllGatherRequest`, that both extends RequestToSync ? Then you don't need to add every params in the base class.

Reply:
I see, though it is only 1 parameter - do you think the complexity introduced in the request checking is worth the verbosity?

Reply:
They are different message types, thus we should separate them.